PowerShell/JumpCloud Module/Docs/Get-JCSystemApp.md

```diff
@@ -0,0 +1,159 @@
+---
+external help file: JumpCloud-help.xml
+Module Name: JumpCloud
+online version: https://github.com/TheJumpCloud/support/wiki/
+schema: 2.0.0
+---
+
+# Get-JCSystemApp
+
+## SYNOPSIS
+Returns the System Apps installed to a JumpCloud System(s)
+
+## SYNTAX
+
+### All (Default)
+```
+Get-JCSystemApp [-SystemID <String>] [-SystemOS <String>] [-SoftwareName <String>] [-SoftwareVersion <String>]
+ [<CommonParameters>]
+```
+
+### Search
+```
+Get-JCSystemApp [-SystemID <String>] [-SystemOS <String>] [-SoftwareName <String>] [-SoftwareVersion <String>]
+ [-Search] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Get-JCSystem app function enables admins to search what software is on their systems
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Get-JCSystemApp -SystemId 'SystemId'
+```
+
+Returns the applications installed in the system with the given -SystemId
+
+### Example 2
+```powershell
+PS C:\> Get-JCSystemApp -SystemOs 'MacOs'
+```
+
+Returns the 'MacOs' systems and all the applications installed for each system
+
+### Example 3
+```powershell
+PS C:\> Get-JCSystemApp -SystemOs 'MacOs' -SoftwareName 'JumpCloud-Agent'
+```
+
+### Example 4
+```powershell
+PS C:\> Get-JCSystemApp -SystemOs 'MacOs' -SoftwareName 'JumpCloud-Agent' -SoftwareVersion '1.12.5'
+```
+
+Returns the 'MacOs' systems that have a 'JumpCloud Agent' application with the version '1.12.5'
+
+### Example 5
+```powershell
+PS C:\> Get-JCSystemApp -SoftwareName 'jumpcloud-agent' -Search
+```
+
+Returns any 'jumpcloud-agent' software installed in all the os systems
+
+## PARAMETERS
+
+### -Search
+Global search ex.
+(1.1.2)
+
+```yaml
+Type: System.Management.Automation.SwitchParameter
+Parameter Sets: Search
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -SoftwareName
+The name of the application you want to search for ex.
+(JumpCloud-Agent, Slack)
```

Comment:
This is a decision point all of us need to make before we move forward...

for macOS Systems: `softwareName` refers to the `bundle_name` but we can technically refer to any field in the SI table. Not only do we need to decide which "name" parameter we want to query but we need to document this  so customers know what value will technically return valid results for `softwareName` parameters on windows/ macOS/ linux.


![Screen Shot 2023-01-26 at 10 50 43 AM](https://user-images.githubusercontent.com/54448601/214911161-f381a96a-96a8-4aba-b6f6-9d30d61904ba.png)

Reply:
Based off of what Joe said, I think the `BundleExecutable` or `Name` field would be the likely candidate for finding the desired software app

Reply:
Decided with filtering with property `name` on the macOs